testng-asserts/src/test/java/org/testng/AssertTest.java

```diff
@@ -602,6 +602,16 @@ public void testAssertEqualsDeepSetFail() {
     Assert.assertEqualsDeep(actualSet, expectedSet);
   }
 
+  @Test(description = "GITHUB-3227", expectedExceptions = AssertionError.class)
+  public void testAssertEqualsNoOrderCollection() {
+    var actual = List.of("a", "b", "b");
+    var expected = List.of("a", "b", "c");
```

Comment:
Replace by a,a,b

Reply:
That change wont be a valid test because then both actual and expected will be equal (since the underlying set reduces it) and so no assertion failures will be triggered.

Reply:
That's the point. Is it expected that aab and abb are equals? 
And it could be extended: is it expected that aaaaaab is equals to abbbbbb?
In both case, I think it should be added as case in order to enforce the expectations

Reply:
> That's the point. Is it expected that aab and abb are equals?

They are not supposed to be equal. But my implementation ends up making them equal. The problem is that both `removeAll` and `Set` dont solve the usecase. Looks like the only option is to be able to sort the collections and then try to compare it. I am not sure if the perf implications need to be considered here.

Reply:
You can sort only if the assertion still holds after the initial check.
Otherwise, create a separate issue if you want to handle it later.

Reply:
I relooked at this. I noticed that we have a `assertEqualsNoOrder` method for `Object[]` arrays and it resorts to removing one element at a time instead of doing a `removeAll()`. That way we ensure that we don't do a greedy removal but only 1 element at a time. Please check now